core/src/main/java/com/exadel/aem/backpack/request/impl/RequestAdapterImpl.java

```diff
@@ -157,20 +173,145 @@
+     *
+     * @param parameterMap       Parameters taken from a {@code SlingHttpRequest}
+     * @param field              Field for instantiation
+     * @param validationMessages Reference to the collection of warnings produced by validation routines
+     * @param validationNeeded   Is validation needed or not
+     * @param genericType        Type of the object to instantiate
+     * @param objectsList        list of objects for multifield
+     * @return True when the List of objects has been successfully populated with validated data; otherwise, false
+     */
+    private boolean initMultifieldList(final Map<?, ?> parameterMap,
+                                       final Field field,
+                                       final List<String> validationMessages,
+                                       final boolean validationNeeded,
+                                       final Class<?> genericType,
+                                       final List<Object> objectsList) {
+        List<Field> objectFields = getAllFields(new ArrayList<>(), genericType);
+        String parameterName = getParameterName(field);
+
+        Map<String, Map<?, ?>> multfieldItems = extractMultifieldParameter(parameterMap, parameterName);
+
+        //instantiate list of objects from multifield properties
+        boolean currentObjectValid = true;
+        for (Map.Entry<String, Map<?, ?>> entry : multfieldItems.entrySet()) {
+            Object nestedObject = createDefaultObject(genericType);
+            if (validationNeeded) {
+                currentObjectValid = initValidateFields(entry.getValue(), nestedObject, validationMessages, objectFields);
+                if (currentObjectValid) {
+                    objectsList.add(nestedObject);
+                } else {
+                    currentObjectValid = false;
+                    break;
+                }
+            } else {
+                initFields(entry.getValue(), nestedObject, objectFields);
+                objectsList.add(nestedObject);
+            }
+        }
+        return currentObjectValid;
+    }
+
+    /**
+     * Gets the multifield parameter items from the request parameter map.
+     *
+     * @param parameterMap  {@code Map} with multifield parameters items
+     * @param parameterName Name of the root parameter
+     * @return Map with parameter properties and values
+     */
+    private Map<String, Map<?, ?>> extractMultifieldParameter(final Map<?, ?> parameterMap, final String parameterName) {
+        Map<String, Map<?, ?>> multfieldItems = new HashMap<>();
+        for (Map.Entry<?, ?> entry : parameterMap.entrySet()) {
+            if ((((String) entry.getKey()).startsWith(parameterName + MULTIFIELD_PARAM_SEPARATOR)
+                    && ((String) entry.getKey()).split(MULTIFIELD_PARAM_SEPARATOR).length == 3)) {
+                Object[] params = ((String) entry.getKey()).split(MULTIFIELD_PARAM_SEPARATOR);
```

Comment:
`String[] params` because result of splitting is nothing but string. It will be cast to `Object` automatically if necessary

Reply:
Fixed.